src/components/Pagination/useControlled.ts

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable react-hooks/rules-of-hooks, react-hooks/exhaustive-deps */
+import * as React from 'react'
+
+export interface UseControlledProps<T = unknown> {
+  /**
+   * This prop contains the component value when it's controlled.
+   */
+  controlled: T | undefined
+  /**
+   * The default value when uncontrolled.
+   */
+  default: T | undefined
+  /**
+   * The component name displayed in warnings.
+   */
+  name: string
+  /**
+   * The name of the state variable displayed in warnings.
+   */
+  state?: string
+}
+
+export function useControlled<T = unknown>({
+  controlled,
+  default: defaultProp,
+  name,
+  state = 'value',
+}: UseControlledProps<T>): [T, (newValue: T) => void] {
+  const { current: isControlled } = React.useRef(controlled !== undefined)
+  const [valueState, setValue] = React.useState(defaultProp)
+  const value = isControlled ? controlled : valueState
+
+  if (process.env.NODE_ENV !== 'production') {
+    React.useEffect(() => {
+      if (isControlled !== (controlled !== undefined)) {
+        console.error(
+          [
+            `Material-UI: A component is changing the ${
```

Comment:
Seriously!!!!!!

This is bad, we can't just cut 'n' paste this. 
Firstly, we don't need this as there is a hook in radix (and one in @committed/hooks) that can do this this for us already. ('@radix-ui/react-use-controllable-state')

Secondly, even though Material is Open source this would require credit and a reference to their licence. Also probably breaks Copywright.